libqtile/backend/x11/core.py

```diff
@@ -73,6 +74,39 @@ class ExistingWMException(Exception):
     pass
 
 
+class SelectionStatus(enum.Enum):
+    empty = enum.auto()
+    targets = enum.auto()
+    awaiting = enum.auto()
+    incremental_transfer = enum.auto()
+    available = enum.auto()
+
+
+class Selection:
+    def __init__(self):
+        self.owner = None
+        self.binary_selection = b""
+        self.selection = ""
+        self.status = SelectionStatus.empty
+        self.targets = []
+        self._cur_tar = -1
+
+    @property
+    def current_target(self):
+        return self.targets[self._cur_tar] if self.targets and self._cur_tar >= 0 else "UTF8_STRING"
+
+    # For backwards compatibility
+    def __index__(self, idx):
+        return self.__dict__[idx]
+
+    def __repr__(self):
+        bin_sel = (self.binary_selection[:29] + b"...") if len(self.binary_selection) > 32 else self.binary_selection
+        sel = (self.selection[:29] + "...") if len(self.selection) > 32 else self.selection
+        return "<x11.Selection owner=({}) bin_sel=({}) sel=({}) status=({}) targets=({}) current_target=({})>".format(
+            self.owner, bin_sel, sel, self.status, self.targets, self.current_target
+        )
```

Comment:
Is this required, or was this just helpful for debugging?

Reply:
This is required for debugging, in my view. Without it, the whole selection as a binary string will get put in the log with the log level set to DEBUG. Clogs the console for large selections (such as images, etc.). This is due to hook.fire automatically logging the event.

Reply:
OK sounds good